### PR TITLE
Get methods for course information

### PR DESCRIPTION
**Pull request by xsfar**
Improved getTime and getDay methods, also now both methods work for online courses, the only case that doesn't work fully currently is a few classes that have different times on different days. I'll replace both methods soon with something better soon, since both methods and future get methods will be redundant and the code is basically the same for the methods expect for one line, but for now this can be used.

### CODE REVIEW

**Comment by xsfar** [comment]
I added get methods for all the information on a given course in a given semester. It works for the vast majority of courses. I'll improve it later to work for ever possible course, also will add some fail safes, such as using the getBuilding method on an online class, and formatting the input from the user.

File: ScheduleCreator/DBAdapter.java [hunk 1]
```diff
@@ -16,21 +16,32 @@
...
+    //Regex expression to get time from the current format.
+    final static String getTime = "(?<=([ ]\\b[A-Z]{3}\\b.\\b[0-9]{3}\\b.+ [0-9]{2}\\b)).+?(?=\\b((TR\\b|MW\\b|MWF\\b|M\\b|T\\b|W\\b|R\\b|F\\b|(	 	TBA\\b))))";
+    //Regex expression to get CRN from the current format.
+    final static String getCRN = "[0-9]{5}";
+    //Regex expression to get building from the current format.
+    final static String getBuilding = "(?<=\\b([	](TR\\b|MW\\b|MWF\\b|M\\b|T\\b|W\\b|R\\b|F\\b|(TBA\\b)))\\b).*(?=\\=)";
+    //Regex expression to get instructor from the current format.
+    final static String getInstructor = "(?<=\\=).*(?= )";
+    //Temporary holding place for course information while being parsed for requested information.
+    private static List<String> lines;
```
Comment (ghost) [inline ScheduleCreator/DBAdapter.java:36]: Should this go in the function that uses it?

**Comment by ghost** [comment]
This is really good. So ParseData isn't used at runtime anymore? 

@jamisonv93 @cloud0x9 @toldge here are my thoughts about organization. Only the unedited original course data files like `Spring 2020.txt` from the website should go in `raw`.  All of our generated files should be stored separately. More importantly, we need a class to generate those files so our work is reproducible. 

Have you guys saved the code used to generate files like `spring2020courses.txt`? I might have missed it. We should be able to delete all of those other files, then click "run class" on the DBGenerator (or whatever) class and rebuild them.

@cloud0x9 with the current DBAdapter methods, the section name (abbreviation, course #, section # like "CSC 250 - 01") is used to make calls. This is probably fine, but we need to settle on one interface for DBAdaptor in order to build the GUI. Another option is to nuke ParseData, move your new code into it, and use it as our "translator" class with DBAdaptor as the interface.

Thoughts?

**Comment by ghost** [comment]
One thing we also _could_ do is remove the ability to save user courses and replace it with the generated data as our persistent data store. The first time the program is runs, the DB is generated and saved. This would allow us to remove the generated DB files from version control.

**Comment by wrycode** [comment]
One thing we also _could_ do is remove the ability to save user courses and replace it with the generated data as our persistent data store. The first time the program is runs, the DB is generated and saved. This would allow us to remove the generated DB files from version control.

File: ScheduleCreator/DBAdapter.java (hunk 1 above)
Comment (xsfar) [reply ScheduleCreator/DBAdapter.java:36]: Should it? I initially put it in the method that uses it, but I was trying to follow the style guide, I don't know.

**Comment by xsfar** [comment]
No ParseData doesn't run at runtime, I just made a text file from the results. I don't know about nuking ParseData, since for any future semesters the whole program would be useless if you can't build a database/textfile for that semester for the program to run on, I say we keep it and just decide on an interface for DBAdapter, also "CSC 250 - 01" is the most concise, its also the easiest since that's how it's given directly from the website, there is no formatting done to it, how else would you do it?

File: ScheduleCreator/DBAdapter.java (hunk 1 above)
Comment (ghost) [reply ScheduleCreator/DBAdapter.java:36]: I feel like it should be a local variable, but it doesn't really matter.

**Comment by ghost** [comment]
Ah, ok. So Parsedata is used to generate the databases we're calling at runtime... I can do the changes I'm proposing in a new pull request and let you guys look at it.

> "CSC 250 - 01" is the most concise, its also the easiest since that's how it's given directly from the website, there is no formatting done to it, how else would you do it

Ok, then let's use it!

Can we merge this?

**Comment by xsfar** [comment]
@wrycode yes

**Pull request merged**